File: sms/mock.go
package sms

import (
	"context"
	"fmt"
	"log"
)

type MockSMSProvider struct{}

func NewMockSMSProvider() *MockSMSProvider {
	return &MockSMSProvider{}
}

func (m *MockSMSProvider) SendOTP(ctx context.Context, payload SMSPayload) error {
	log.Printf("[MOCK SMS PROVIDER] Sending OTP '%s' to mobile '%s' for purpose '%s'", payload.OTP, payload.Mobile, payload.Purpose)
	fmt.Printf("\n=======================================================\n")
	fmt.Printf("[MOCK SMS] Mobile: %s | OTP: %s | Purpose: %s\n", payload.Mobile, payload.OTP, payload.Purpose)
	fmt.Printf("=======================================================\n\n")
	return nil
}

func (m *MockSMSProvider) GetProviderName() string {
	return "MOCK"
}

File: backend_api/sms/mock.py
from sms.provider import SMSProvider, SMSPayload

class MockSMSProvider(SMSProvider):
    def send_otp(self, context, payload: SMSPayload):
        print(f"[MOCK SMS PROVIDER] Sending OTP '{payload.otp}' to mobile '{payload.mobile}' for purpose '{payload.purpose}'")
        print("\n=======================================================")
        print(f"[MOCK SMS] Mobile: {payload.mobile} | OTP: {payload.otp} | Purpose: {payload.purpose}")
        print("=======================================================\n")

    def get_provider_name(self) -> str:
        return "MOCK"
